workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/selection/VaultWatcherService.kt

```diff
@@ -104,6 +108,7 @@ class VaultWatcherService(tokenObserver: TokenObserver? = null) : SingletonSeria
     }
 
     private fun addTokenToCache(stateAndRef: StateAndRef<FungibleToken>) {
+        // TODO after some cache size is reached, stop listening to new events and print massive warning & switch to database selection on this event
```

Comment:
Need to make sure we dont exceed max cache size here. should we track current cache size and only add new tokens to cache if cache size < max cache size. Not ideal for account based selection as may result in some accounts not being able to hold any tokens but i suspect this will only manifest for very very "busy" nodes.